### ISSUE TITLE
Apt. number field to listings, and javascript?

### ISSUE THREAD

**Issue by aeggum** [mswysocki/Madison-Student-Apartment-Finder#35]
I'm planning on doing a migration to add this column to the model soon.

One thing I was thinking about with it, then, is that when creating a listing, by default it can be house, but if they select apt. then a field appears for the apartment number (using javascript), and vice versa.  I think that wouldn't be too difficult to add, correct?

Also, then House/Apartment would be one of the first things on the create listing page.

**Comment by cromdenne** [comment]
I like this, I forgot to ask about it.  I don't think it would be too hard to do that with javascript.  And I think it would clean things up a lot when displaying the residence, being able to display the apartment building name rather than the address.

**Comment by xibaca** [comment]
Yeah, I agree on that. I'm sure we can do it with Javascript. We may be careful so when this field appears it actually don't mess with all the other elements that are already in the page, but I think that's the only big concern.

**Comment by aeggum** [comment]
I added the field to the model and it's on the add a listing page now, without the js. 

Tonight I will try to update the current stuff in the database so that apartments have apartment numbers and then do the changes we've talked about.

Do you think I should add another field, say 'Building name', also for use only with apartments?  Then for apts as you say we can display only as the main name. 
- If so, how should we handle building names? (ie. choose from a list pre-populated, or anything goes, etc)

**Comment by xibaca** [comment]
Humm.. handling building names is a bit complicated, as pretty much anything can be the name and having a extensive list of building names is unfeasible..

We can accept pretty much anything and eventually we may need to have a way of detecting innappropriate names (and any kind of innappropriate content throughtout the website, as well).

I had an idea now, speaking of adding fields. What about a "reference" field, so that people could write an easy way of locating that listing (e.g. "right by Fresh Market", "near Brats")?

**Comment by aeggum** [comment]
I think another field for reference might just add even more ambiguity.

**Comment by cromdenne** [comment]
I do like the building name idea.  Building name would essentially be the same as the "address" field for houses, since those are what would be displayed for each residence.  So maybe we organize it like the first query when a person adds a residence should be "house or apartment" and then if they choose house, just allow them to enter the address, but if they choose apartment, ask them for the building name and the address, and then the apartment number separately also.  Then for houses, we could just make the building name and apartment number fields null (unless that's not general enough, then I guess we could make apartments and houses separate objects, but that might also be a lot to do at this point).

I think building name is a good idea for now though.  As Guilherme says, it might be complicated moderating that in the future, but I think that's something we should just worry about later, because I think it's much more helpful to a person searching residences to see the name of the apartment building than just the address.

**Comment by aeggum** [comment]
Alright sounds good.  Tonight I'll try to add building names to the model.   I think it should be fine if we set them to null.   I think there is a way to do dynamic validations, so hopefully it won't be too tough.  
Separate models seems like overkill, or at least overkill for the next two weeks.  It would break pretty much everything we have to do with listings so far.

**Comment by xibaca** [comment]
Nice. I was just wondering if the reference field would be a good idea. I guess if this turn out to be something people find necessary, they could comment on the listing anyways, so yeah, that's true.

I agree with you guys about the decisions above, and I think it is wise to avoid as much changes in the database as possible, you're right.

**Comment by aeggum** [comment]
For apartments, the apt. numbers and building name are now required.  And addresses are don't need to be unique for apartments. 
The changes should be up on Heroku.
